demo/react-hooks.html

```diff
@@ -0,0 +1,94 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="UTF-8" />
+  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+  <title>Gridstack.js React integration example</title>
+  <link rel="stylesheet" href="demo.css" />
+  <script src="../dist/gridstack-h5.js"></script>
+
+  <!-- Scripts to use react inside html -->
+  <script src="https://unpkg.com/react@16/umd/react.production.min.js"></script>
+  <script src="https://unpkg.com/react-dom@16/umd/react-dom.production.min.js"></script>
+  <script src="https://unpkg.com/babel-standalone@6.15.0/babel.min.js"></script>
+</head>
+
+<body>
+  <div id="root"></div>
+</body>
+
+<script type="text/babel">
+  const { useState, useEffect, createRef, useRef } = React
+
+  const Item = ({ id }) => <div>I am item: {id}</div>
+
+  const ControlledStack = ({ items, addItem }) => {
+    const refs = useRef({})
+
+    if (Object.keys(refs.current).length !== items.length) {
+      items.forEach(({ id }) => {
+        refs.current[id] = refs.current[id] || createRef()
+      })
+    }
+
+    useEffect(() => {
+      const grid = GridStack.init()
+      grid.removeAll(false)
+      items.forEach(({ id }) => {
+        grid.makeWidget(refs.current[id].current)
+      })
+    }, [items])
+
+    return (
+      <div>
+        <button onClick={addItem}>Add new widget</button>
+        <div className="grid-stack">
+          {items.map((item, i) => {
+            return (
+              <div
+                ref={refs.current[item.id]}
+                key={item.id}
+                className={'grid-stack-item'}
+                gs-w="12"
```

Comment:
don't think you want entire row widget

Reply:
My intention here is that as a user adds items to the array (i.e. updates the database), that these new items come through to the client and get added to the bottom of the grid stack as rows. Users can then re-adjust the layout to incorporate those rows.

Is there a better way of achieving this?

Reply:
Ah, I see that I would need to recreate the grid from some kind of state, otherwise grid items will always be full rows on first component render

Reply:
you can serialize the x,y,w,h of each grid item. adding them to the bottom and making when a full row is just weird. Look at view adding a single widget but you say yours come from backend as complete list. Which may or may not be what you want in general. I would suggest showing both cases.

Reply:
I've updated the example to include the old react example, re-written using functional components and also the controlled example